service/src/test/java/com/bankbazaar/service/service/TripActivityServiceTest.java

```diff
@@ -0,0 +1,126 @@
+import com.bankbazaar.dto.model.TripDto;
+import com.bankbazaar.dto.model.UserDto;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@Slf4j
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles({"test"})
+class TripActivityServiceTest {
+
+    @Autowired
+    UserService userService;
+
+    @Autowired
+    TripService tripService;
+
+    @Autowired
+    TripActivityService service;
+
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
+    private TripRepository tripRepository;
+    @Autowired
+    private TripUserMapRepository tripUserMapRepository;
+    @Autowired
+    private TripActivityRepository tripActivityRepository;
+
+    @BeforeEach
+    void setUp() throws ParseException {
+        log.info("Executing cleanup");
+        tripUserMapRepository.deleteAll();
+        tripActivityRepository.deleteAll();
+        userRepository.deleteAll();
+        tripRepository.deleteAll();
+
+        UserDto userDto =new UserDto();
+        userDto.setUserName("name");
+        userDto.setPassword("password");
+        Date date = new SimpleDateFormat("yyyy-MM-dd").parse("1998-05-23");
+        userDto.setDob(date);
+        userDto.setEmail("name@mail.com");
+        userDto.setPhone("1234567891");
+        UserDto user = userService.addUser(userDto);
+
+        TripDto tripDto =new TripDto();
+        tripDto.setTripName("name");
+        tripDto.setTripDescription("trip");
+        tripDto.setDestination("trip");
+        Date startDate = new SimpleDateFormat("yyyy-MM-dd").parse("1998-05-23");
+        tripDto.setStartDate(startDate);
+        Date endDate = new SimpleDateFormat("yyyy-MM-dd").parse("1998-05-23");
+        tripDto.setEndDate(endDate);
+        tripService.addTrip(tripDto, user.getUserId());
+    }
+
+    @Test
+    void addTripActivity() throws ParseException {
+        UserEntity user = userRepository.findByEmail("name@mail.com").get();
+        List<TripUserMapEntity> tripUserMap = tripUserMapRepository.findAll();
+        TripActivityDto tripActivityDto = new TripActivityDto();
+        tripActivityDto.setActivityStatus("PENDING");
+        tripActivityDto.setActivityDescription("trip activity");
+        Date date = new SimpleDateFormat("yyyy-MM-dd").parse("1998-05-23");
+        tripActivityDto.setActivityTime(date);
+        tripActivityDto.setLocation("park");
+
+        service.addTripActivity(tripUserMap.get(0).getTripId(),tripActivityDto,user.getUserId());
+        assertEquals(tripActivityRepository.count(),1);
```

Comment:
2 more tests are required here.
1. Validate that the tripActivityRepository contains 0 elements initially
2.  addTripActivity returns a dto. Get the id from the dto and fetch from db and validate all the fields of response, db and request are same